directory-menu@torchipeppo/files/directory-menu@torchipeppo/applet.js

```diff
@@ -36,15 +49,21 @@ class CassettoneApplet extends Applet.IconApplet {
         this.settings.bind("starting-uri", "starting_uri", this.normalize_tilde, this.starting_uri);
         this.settings.bind("show-hidden", "show_hidden", null, null);
         this.settings.bind("icon-name", "icon_name", this.set_applet_icon_symbolic_name, null);
-        this.settings.bind("tooltip", "tooltip_text", (newtext) => {this.set_applet_tooltip(_(newtext))}, null);
+        this.settings.bind("label", "label", this.set_applet_label, null);
+        this.settings.bind("tooltip", "tooltip_text", (txt) => this.set_applet_tooltip(_(txt)), null);
+        this.settings.bind("limit-characters", "limit_characters", null, null);
         this.settings.bind("character-limit", "character_limit", null, null);
         this.starting_uri = this.normalize_tilde(this.starting_uri);
 
-        this.set_applet_icon_symbolic_name(this.icon_name);
         this.set_applet_tooltip(_(this.tooltip_text));
+        this.set_applet_icon_symbolic_name(this.icon_name);
+        this.set_show_label_in_vertical_panels(false);
+        this.set_applet_label(this.label)
 
         this.actor.connect('enter-event', Lang.bind(this, this.on_enter_event));
         this.actor.connect('button-release-event', Lang.bind(this, this.on_button_release_event));
+
+		this.setAllowedLayout(Applet.AllowedLayout.BOTH);
```

Comment:
I still don't understand the "improved vertical panels" thing. Is it this line?

Reply:
Yes, generally, when you moved the applet to a vertical panel it would alert to the user that it might not work well there, but for what I have tested it works without any issues in a vertical panel, also theres one more line above that disables the text on vertical panels if set.